baking/src/tezos_baking/tezos_setup_wizard.py

```diff
@@ -74,14 +74,90 @@
 """
 
 
-def fetch_snapshot(url):
+def fetch_snapshot(url, sha256=None):
+
+    dirname = TMP_SNAPSHOT_LOCATION
+    filename = os.path.join(dirname, "octez_node.snapshot")
+    metadata_file = os.path.join(dirname, "octez_node.snapshot.sha256")
+
+    # updates or removes the 'metadata_file' containing the snapshot's SHA256
+    def dump_metadata(metadata_file=metadata_file, sha256=sha256):
+        if sha256:
+            with open(metadata_file, "w+") as f:
+                f.write(sha256)
+        elif os.path.exists(metadata_file):
+            os.remove(metadata_file)
```

Comment:
This is not the best solution to [my previous comment](https://github.com/serokell/tezos-packaging/pull/661#discussion_r1267973197) because it's possible for `os.remove` to still throw an exception here.

The fact that `metadata_file` existed when `os.path.exists` is called does not guarantee that it still exists when `os.remove` is called.

So, as much as it is unlikely to happen and even tho it's uglier to see in code, let's use exception handling